pip/compat.py

```diff
@@ -41,7 +46,11 @@ def console_to_str(s):
         try:
             return s.decode(sys.__stdout__.encoding)
         except UnicodeDecodeError:
-            return s.decode('utf_8')
+            if WINDOWS:
+                from ctypes import cdll
+                return s.decode("cp" + str(cdll.kernel32.GetACP()), 'replace')
```

Comment:
Yeah I think it doesn't matter that much too.
After such a long discussion, I think we ended up with some solutions together.
1.  if windows, `return s.decode("cp" + str(cdll.kernel32.GetACP()), 'replace')` with try-catch for `ImportError` of ctypes.
2.  if windows, `return s.decode("cp" + str(cdll.kernel32.GetOEMCP()), 'replace') `  with try-catch for `ImportError` of ctypes.
3. Regardless of whether it is windows or not, use `s.decode(locale.getpreferredencoding(), 'replace')`

I prefer the third choice for now. It's more clean and no need to handle `ImportError`
If you think there's better way just feel free to let me know, or I will try to change it to the third solution.
I wonder if is OK with you?